drivers/sensor/hts221/hts221.c

```diff
@@ -180,17 +196,94 @@ int hts221_init(const struct device *dev)
 	return 0;
 }
 
-static struct hts221_data hts221_driver;
-static const struct hts221_config hts221_cfg = {
-	.i2c_bus = DT_INST_BUS_LABEL(0),
-	.i2c_addr = DT_INST_REG_ADDR(0),
-#if HTS221_TRIGGER_ENABLED
-	.drdy_pin = DT_INST_GPIO_PIN(0, drdy_gpios),
-	.drdy_flags = DT_INST_GPIO_FLAGS(0, drdy_gpios),
-	.drdy_controller = DT_INST_GPIO_LABEL(0, drdy_gpios),
-#endif /* HTS221_TRIGGER_ENABLED */
-};
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "HTS221 driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macros
+ */
+
+#define HTS221_DEVICE_INIT(inst)					\
+	DEVICE_DT_INST_DEFINE(inst,					\
+			    hts221_init,				\
+			    hts221_pm_control,				\
+			    &hts221_data_##inst,			\
+			    &hts221_config_##inst,			\
+			    POST_KERNEL,				\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &hts221_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_HTS221_TRIGGER
+#define HTS221_CFG_IRQ(inst)					\
+	.gpio_drdy = GPIO_DT_SPEC_INST_GET(inst, irq_gpios)
+#else
+#define HTS221_CFG_IRQ(inst)
+#endif /* CONFIG_HTS221_TRIGGER */
+
+#define HTS221_SPI_OPERATION (SPI_WORD_SET(8) |				\
+			      SPI_OP_MODE_MASTER |			\
+			      SPI_MODE_CPOL |				\
+			      SPI_MODE_CPHA |				\
+			      SPI_HALF_DUPLEX)				\
+
+#define HTS221_CONFIG_SPI(inst)						\
+	{								\
+		.ctx = {						\
+			.read_reg =					\
+			   (stmdev_read_ptr) stmemsc_spi_read,		\
+			.write_reg =					\
+			   (stmdev_write_ptr) stmemsc_spi_write,	\
+			.handle =					\
+			   (void *)&hts221_config_##inst.stmemsc_cfg,	\
+		},							\
+		.stmemsc_cfg.spi = {					\
+			.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),	\
+			.spi_cfg = SPI_CONFIG_DT_INST(inst,		\
+					   HTS221_SPI_OPERATION,	\
+					   0),				\
```

Comment:
Here is better to use SPI_DT_SPEC_INST_GET() and, similarly, I2C_DT_SPEC_INST_GET() for i2c bus. I'm saying so because stmemsc is going to change i/f because of #38343, which is not merged yet, so, depending which one will be merged first the other should be aligned. Not sure what the best option is. If this one will be merged first I will then change my PR, otherwise you can align this one from now (and cherry-pick my commit c74640bceee66fbfa87bc019b23936225d4da2bf, which can be merged with this PR instead of mine). What do you think?

Reply:
Maybe @gmarull and @mbolivar can provide their feedback about how to proceed.

Reply:
You PR will probably land for the next release, same for this PR as it has to go through API change process. So there will be time for adapting this PR against the changes on stmemsc.

Reply:
@avisconti change was applied btw.

Reply:
> @avisconti change was applied btw.

OK, I'm reviewing right now